diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 49b8a9a..939eb60 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -1639,7 +1639,7 @@ static void try_preempt(struct task_struct *p, struct rq *this_rq)
 
 	cpumask_and(&tmp, &cpu_online_map, &p->cpus_allowed);
 
-	for (i = 0; i < num_possible_cpus(); i++) {
+	for (i = 0; i < num_online_cpus(); i++) {
 		struct rq *rq = this_rq->cpu_order[i];
 
 		if (!cpumask_test_cpu(rq->cpu, &tmp))
@@ -1900,7 +1900,7 @@ static inline int select_best_cpu(struct task_struct *p)
 			return rq->cpu;
 	}
 
-	for (i = 0; i < num_possible_cpus(); i++) {
+	for (i = 0; i < num_online_cpus(); i++) {
 		struct rq *other_rq = task_rq(p)->cpu_order[i];
 		int entries;
 
