diff --git a/mm/zswap.c b/mm/zswap.c
index cd91fd9d96b8..bcddebfe9c57 100644
--- a/mm/zswap.c
+++ b/mm/zswap.c
@@ -82,7 +82,7 @@
 module_param_cb(enabled, &zswap_enabled_param_ops, &zswap_enabled, 0644);
 
 /* Crypto compressor to use */
-#define ZSWAP_COMPRESSOR_DEFAULT "lzo"
+#define ZSWAP_COMPRESSOR_DEFAULT "lz4"
 static char *zswap_compressor = ZSWAP_COMPRESSOR_DEFAULT;
 static int zswap_compressor_param_set(const char *,
 				      const struct kernel_param *);
--
1.0


