From 820bfe3fc1a6ffddec9019bf80d6dbd541bdd10b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sveinar=20S=C3=B8pler?= <cybermax@dexter.no>
Date: Sun, 17 Jan 2021 13:53:54 +0100
Subject: [PATCH] Version

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index e99310c2b..92c0f623e 100644
--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,10 @@ $(if $(filter __%, $(MAKECMDGOALS)), \
 PHONY := __all
 __all:
 
+CYBVERSION = -cybmod13
+CYBNAME = CacULE Powered
+EXTRAVERSION := $(EXTRAVERSION)$(CYBVERSION)
+
 # We are using a recursive build, so we need to do a little thinking
 # to get the ordering right.
 #
-- 
2.25.1

